### PR TITLE
fix: handle "\0" in `chunk.modules` object key

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

- Workaround for https://github.com/rolldown/rolldown/issues/1115

I was looking into this, but It looks like napi-rs just introduced `node_api_create_property_key_utf8` for js object key conversion in https://github.com/napi-rs/napi-rs/pull/2358 (though it's gated under experimental features and node version detection).

I'm not sure if this change will be needed but perhaps some tests are nice to have, so just preparing a draft PR here.

### CODE REVIEW

**Comment by hyf0** [comment]
CI/Tests looks fine. Seems we solve this problem.

**Comment by hi-ogawa** [comment]
This broke js side `chunk.modules` typing, so I need to figure that out.

**Comment by hi-ogawa** [comment]
I update the implementation to use custom `to_napi_value`. When napi-rs eventually supports this by default, we can remove the wrapper struct `BindingChunkModules`.

File: crates/rolldown_binding/src/types/binding_rendered_chunk.rs [hunk 1]
```diff
@@ -34,13 +39,48 @@ impl From<rolldown_common::RollupRenderedChunk> for RenderedChunk {
...
-        .collect(),
+      modules: BindingChunkModules::new(value.modules),
       imports: value.imports.iter().map(|x| x.to_string()).collect(),
       dynamic_imports: value.dynamic_imports.iter().map(|x| x.to_string()).collect(),
     }
   }
 }
+
+// use own map wrapper to workaround "\0" issue
+// https://github.com/napi-rs/napi-rs/blob/f116eaf5e54090db4dca8a00ccdb684543a39e86/crates/napi/src/bindgen_runtime/js_values/map.rs#L26
```
Comment (hyf0) [inline crates/rolldown_binding/src/types/binding_rendered_chunk.rs:50]: Wonder if we should enable napi's `experimental` feature

https://github.com/napi-rs/napi-rs/blob/f116eaf5e54090db4dca8a00ccdb684543a39e86/crates/napi/src/bindgen_runtime/js_values/map.rs#L31-L38C11

So it solves the problem under the hood.

@Brooooooklyn cc

**Comment by hyf0** [comment]
I'm gonna keep https://github.com/rolldown/rolldown/issues/1115 open, and consider this PR as a workaround.

**Review by hyf0 — APPROVED** [review]

**Pull request merged**